### ISSUE TITLE
More unit test failures -- new environment

### ISSUE THREAD

**Issue by bobhy** [xonsh/xonsh#3399]
Trying to get a clean unit test run on unmodified master before I dive into an enhancement.  
Was able to fix one failure, but am still stuck with these:

## Current Behavior
`py.test -q`

### Traceback (if applicable)

<details>

```
=================================== FAILURES ===================================
________________________________ test_decorator ________________________________

    def test_decorator():
>       check_stmts("@g\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 2 != 1

tests/test_parser.py:1931: AssertionError
_______________________________ test_decorator_2 _______________________________

    def test_decorator_2():
>       check_stmts("@h\n@g\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 3 != 1

tests/test_parser.py:1935: AssertionError
_____________________________ test_decorator_call ______________________________

    def test_decorator_call():
>       check_stmts("@g()\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 2 != 1

tests/test_parser.py:1939: AssertionError
___________________________ test_decorator_call_args ___________________________

    def test_decorator_call_args():
>       check_stmts("@g(x, y=10)\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 2 != 1

tests/test_parser.py:1943: AssertionError
_________________________ test_decorator_dot_call_args _________________________

    def test_decorator_dot_call_args():
>       check_stmts("@h.g(x, y=10)\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 2 != 1

tests/test_parser.py:1947: AssertionError
_______________________ test_decorator_dot_dot_call_args _______________________

    def test_decorator_dot_dot_call_args():
>       check_stmts("@i.h.g(x, y=10)\ndef f():\n  pass", False)
E       AssertionError: Ast nodes do not have the same line number : 2 != 1

tests/test_parser.py:1951: AssertionError
=============================== warnings summary ===============================
tests/test_mpl.py::test_mpl_preserve_font_size
tests/test_mpl.py::test_mpl_preserve_face_color
tests/test_mpl.py::test_mpl_preserve_width
tests/test_mpl.py::test_mpl_preserve_height
tests/test_mpl.py::test_mpl_preserve_dpi
tests/test_mpl.py::test_mpl_preserve_image_tight
tests/test_mpl.py::test_mpl_preserve_standard
  /mnt/myd/dev/src/xonsh/venv/lib/python3.8/site-packages/matplotlib/axes/_base.py:3270: DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    left, right = sorted([left, right], reverse=reverse)

tests/test_mpl.py::test_mpl_preserve_font_size
tests/test_mpl.py::test_mpl_preserve_face_color
tests/test_mpl.py::test_mpl_preserve_width
tests/test_mpl.py::test_mpl_preserve_height
tests/test_mpl.py::test_mpl_preserve_dpi
tests/test_mpl.py::test_mpl_preserve_image_tight
tests/test_mpl.py::test_mpl_preserve_standard
  /mnt/myd/dev/src/xonsh/venv/lib/python3.8/site-packages/matplotlib/axes/_base.py:3652: DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    bottom, top = sorted([bottom, top], reverse=reverse)

tests/test_parser.py::test_is
  <test-ast>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?

tests/test_parser.py::test_is_not
  <test-ast>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?

-- Docs: https://docs.pytest.org/en/latest/warnings.html
6 failed, 3960 passed, 15 skipped, 16 warnings in 33.18s
```

</details>

**Comment by gforsyth** [comment]
Hey @bobhy  -- thanks for reporting! I have fixes for the 3.8 failures in #3384 if you want to try pulling those down

**Comment by bobhy** [comment]
Yep, I can confirm #3384 fixes these errors for me.
